lib/Cro/WebApp/Template/AST.pm6

```diff
@@ -331,8 +332,31 @@ sub template-part-args(Str $name) {
     orwith %*CRO-TEMPLATE-EXPLICIT-PARTS{$name} {
         $_
     }
+    orwith find-template-part-provider($name) -> &provider {
```

Comment:
Maybe it doesn't actually make sense, but wouldn't you expect a template part provider to take precedence over a `CRO-TEMPLATE-EXPLICIT-PARTS`, and pass the arguments to said provider instead?

Reply:
In code this is something like:

```
route {
    template-part 'foo', -> { 'implicit' }
    get -> {
        template 'foo.crotmp', parts => { foo => 'explicit' }
    }
}
```

Since the explicit one is more local / nested than the part provider, I think it's right this way around.

Reply:
Yes, I agree with that. The idea I had was that passing arguments to a `template-part` subroutine might allow for more flexibility:


```raku
route {
    template-part 'foo', Str $argument -> { $argument ~~ /^\w+d/ ?? $argument !! $argument.flip }
    get -> {
        template 'foo.crotmp', parts => { foo => 'explicit' }
    }
}
```

Maybe that's too convoluted, maybe that's not useful, maybe that's confusing.

Reply:
I'd considered parametric template parts, but decided not to do that for now. It's a step up in power, but also in complexity, and I'm not yet sure if it's justified. Probably one of those things I'd rather see being missed before adding it.